### PR TITLE
4950-Add a new test for withUnixLineEndings

### PR DESCRIPTION
**Pull request by angelaclan**
StirngLineEndingTest Class added.
It does test line endings in general, not only unixLineEndings, but unixLineEndings are included.

tests Line Endingless String to Be Equals
tests origin string to be reproduced From Target String And OriginLineEnding
tests TargetLine does not contain OriginLineEnding .
The test case is a subclass of parametrized tests, and it provides a matrix with the folowing cases 
 #targetLineEnding addOptions:
			{String cr.
			String lf.
			String crlf};
 #originLineEnding addOptions:
			{String cr.
			String lf.
			String crlf};
#templateString addOptions:
			{'I am a String{1}' . 
			'{1}I am a String' . 
			'I am a{1}String'.
			'{1}I am{1}a String{1}'};
The test case adds 108 new tests to the suit of string tests .

### CODE REVIEW

**Comment by MarcusDenker** [comment]
changed the tittle to be nicer

**Comment by guillep** [comment]
I restarted the CI

File: src/Collections-Tests/StringLineEndingsTest.class.st [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+{ #category : #'as yet unclassified' }
+StringLineEndingsTest >> testOriginStringCanBeProducedFromTargetStringAndOriginalLineEnding [
+	
+	| testingSubjectWithTargetLineEndingTransformedWithOriginalLineEnding |
+	testingSubjectWithTargetLineEndingTransformedWithOriginalLineEnding := testingSubjectWithTargetLineEnding withLineEndings: originLineEnding .
+	self assert: testingSubjectWithTargetLineEndingTransformedWithOriginalLineEnding equals: testingSubjectWithOriginLineEnding 
+	
+	" transformedString := 'I'm a string <<lf>>' withLineEnding: <<cr>> 
+	>>> assert: I'm a string <<cr>> (transformedString) equals: I'm a string <<cr>> ( subjectWithOriginLineEnding )"
+]
```
Comment (guillep) [inline src/Collections-Tests/StringLineEndingsTest.class.st:82]: Is this comment a left over?

File: src/Collections-Tests/StringLineEndingsTest.class.st [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		forSelector: #targetLineEnding addOptions: {String cr . String lf . String crlf};
+		forSelector: #originLineEnding addOptions: {String cr . String lf . String crlf};
+		forSelector: #templateString addOptions: {'I am a String{1}'. '{1}I am a String' . 'I am a{1}String'. '{1}I am{1}a String{1}'};
+		
+		
+		
+		yourself
+		
+	
+
```
Comment (guillep) [inline src/Collections-Tests/StringLineEndingsTest.class.st]: Why so many empty lines?

**Review by guillep — CHANGES_REQUESTED** [review]
It would be nice to fix these little details before integration.

I'll make sure myself that the issue introduces no dependencies as soon as we have a green build.

**Comment by angelaclan** [comment]
Le dim. 15 mars 2020 à 14:36, Guille Polito <notifications@github.com> a
écrit :

> *@guillep* requested changes on this pull request.
>
> It would be nice to fix these little details before integration.
>
> I'll make sure myself that the issue introduces no dependencies as soon as
> we have a green build.
> ------------------------------
>
> In src/Collections-Tests/StringLineEndingsTest.class.st
> <https://github.com/pharo-project/pharo/pull/5811#discussion_r392674426>:
>
> > +		copyReplaceAll: originLineEnding with: ''.
> +	targetWithoutLineEnding := testingSubjectWithTargetLineEnding
> +		copyReplaceAll: targetLineEnding with: ''.
> +	self assert: originWithoutLineEnding equals: targetWithoutLineEnding
> +]
> +
> +{ #category : #'as yet unclassified' }
> +StringLineEndingsTest >> testOriginStringCanBeProducedFromTargetStringAndOriginalLineEnding [
> +	
> +	| testingSubjectWithTargetLineEndingTransformedWithOriginalLineEnding |
> +	testingSubjectWithTargetLineEndingTransformedWithOriginalLineEnding := testingSubjectWithTargetLineEnding withLineEndings: originLineEnding .
> +	self assert: testingSubjectWithTargetLineEndingTransformedWithOriginalLineEnding equals: testingSubjectWithOriginLineEnding
> +	
> +	" transformedString := 'I'm a string <<lf>>' withLineEnding: <<cr>>
> +	>>> assert: I'm a string <<cr>> (transformedString) equals: I'm a string <<cr>> ( subjectWithOriginLineEnding )"
> +]
>
> Is this comment a left over?
>
  it was a comment to help clearify what this piece of code is doing since
the vars are long. if its more confusing then can remove it. or should move
to the top?

------------------------------

**Comment by angelaclan** [comment]
i found the most update version is not saved!!! let me know what is
better to do for the comment i can change both issues and do pull request
again

**Comment by guillep** [comment]
> " transformedString := 'I'm a string <<lf>>' withLineEnding: <<cr>> + >>> assert: I'm a string <<cr>> (transformedString) equals: I'm a string <<cr>> ( subjectWithOriginLineEnding )" +] Is this comment a left over?
> it was a comment to help clearify what this piece of code is doing since the vars are long. if its more confusing then can remove it. or should move to the top?

The problem is that it looks like normal code that was commented :).
It would be nice, instead, to have a written explanation in english around it to explain what it means. Because we don't get it by just reading it ;)

> better to do for the comment i can change both issues and do pull request again

As you please, you can also check with @sbragagnolo , he may help you in updating it.

Also, I realised that the code also has uncategorised methods ("as yet unclassified").
Test methods should be categorised as "test". Helpers as "helpers".

Thanks Angela :)

**Comment by angelaclan** [comment]
Hi, the changes are done. let me know if its good. Thanks!

**Review by guillep — APPROVED** [review]
Thanks!

**Pull request merged**